backend/courses/util.py

```diff
@@ -712,3 +712,56 @@ def get_semesters(semesters: str = None) -> list[str]:
             if s not in possible_semesters:
                 raise ValueError(f"Provided semester {s} was not found in the db.")
     return sorted(semesters)
+
+
+def historical_year_probability(current, courses):
+    """
+    current: the current semester represented in the 20XX(A|B|C) format
```

Comment:
Super nitpicky: we usually use a specific format for these following the sphinx format